pkg/monitor/azure/nsg/nsg.go

```diff
@@ -121,12 +123,21 @@ func (n *NSGMonitor) Monitor(ctx context.Context) []error {
 	workerProfiles, _ := api.GetEnrichedWorkerProfiles(n.oc.Properties)
 	workerSubnets := make([]subnetNSGConfig, 0, len(workerProfiles))
 	workerPrefixes := make([]netip.Prefix, 0, len(workerProfiles))
+	subnetSet := map[string]struct{}{}
 	for _, wp := range workerProfiles {
 		// Customer can configure a machineset with an invalid subnet.
 		// In such case, the subnetID will be empty.
-		if len(wp.SubnetID) == 0 {
+		//
+		// We do not need to consider profiles with 0 machines.
+		if len(wp.SubnetID) == 0 || wp.Count == 0 {
+			continue
+		}
+
+		// Many profiles can have the same subnet ID.  To minimize the possibility of throttling, we only get it once.
+		if _, ok := subnetSet[wp.SubnetID]; ok {
 			continue
 		}
+		subnetSet[wp.SubnetID] = struct{}{}
 
 		s, err := n.toSubnetConfig(ctx, wp.SubnetID)
 		if err != nil {
```

Comment:
I'm in favor of this change as well, though I imagine it would be a larger refactor than this PR is currently so I'd also be ok with doing that as a follow-up item rather than in this PR if necessary.

Reply:
Sorry for my late reply but I'm currently on a family emergency leave this week.  Can we do this as a follow up instead and I'll make sure it'll be addressed by the next release.

Reply:
I'm fine with that. In fact I'm looking for things to do in Go so if you like we can "switch" and I can have a go ( :wink: ) at doing the refactor?

Reply:
@mociarain sorry I just saw your reply.. I already pushed a new PR (#3358) for this, but feel free to comment.  Let me know if you have any comments about it.